recipes/yarp/meta.yaml

```
@@ -0,0 +1,130 @@
+{% set name = "yarp" %}
+{% set namecxx = "yarp-cxx" %}
+{% set namepython = "yarp-python" %}
+{% set version = "3.7.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/robotology/yarp/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 3b2275e0cc17ce41e44f6ae15adfc76e49a61f5ead8a6dd42ce2cd88fffdcbed
+
+build:
+  number: 0
+
+outputs:
+  - name: {{ namecxx }}
+    script: build_cxx.sh  # [unix]
+    script: bld_cxx.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage(namecxx, max_pin='x.x.x') }}
```

```suggestion
        - {{ pin_subpackage(namecxx) }}
```
This is a very strict pinning. Does the upstream project not follow SEMVER? You already have "exact" pinning of yarp-cxx in the yarp-python. Do external projects need to pin to the minor version?

---

Upstream follows SEMVER (or similar) for API, but does not have any kind of ABI stability, so for example a public struct can get a new non trailing attributes in a patch release (or any other ABI-breaking change, see https://community.kde.org/Policies/Binary_Compatibility_Issues_With_C%2B%2B#The_Do.27s_and_Don.27ts).